src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/UpdateContactDetailsContestedSubmittedHandler.java

```diff
@@ -0,0 +1,137 @@
+            return handleNoticeOfChangeWorklow(callbackRequest, userAuthorisation);
+        }
+
+        if (callbackRequest.getCaseDetailsBefore().getData().isApplicantRepresentedByASolicitor()
+            != callbackRequest.getCaseDetails().getData().isApplicantRepresentedByASolicitor()
+            || callbackRequest.getCaseDetailsBefore().getData().isRespondentRepresentedByASolicitor()
+            != callbackRequest.getCaseDetails().getData().isRespondentRepresentedByASolicitor()) {
+            return handleNoticeOfChangeWorklow(callbackRequest, userAuthorisation);
+        }
+
+        persistOrgPolicies(caseData, callbackRequest.getCaseDetailsBefore().getData());
+
+        return GenericAboutToStartOrSubmitCallbackResponse.<FinremCaseData>builder()
+            .data(finremCaseDetails.getData()).build();
+    }
+
+    private GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handleNoticeOfChangeWorklow(
+        FinremCallbackRequest request, String userAuthorisation) {
+        CaseDetails caseDetails = finremCaseDetailsMapper.mapToCaseDetails(request.getCaseDetails());
+        CaseDetails originalCaseDetails = finremCaseDetailsMapper.mapToCaseDetails(request.getCaseDetailsBefore());
+        AboutToStartOrSubmitCallbackResponse nocResponse = nocWorkflowService.handleNoticeOfChangeWorkflow(
+            caseDetails, userAuthorisation, originalCaseDetails);
+
+        CaseType caseType = request.getCaseDetails().getCaseType();
+        return GenericAboutToStartOrSubmitCallbackResponse.<FinremCaseData>builder()
+            .data(finremCaseDetailsMapper.mapToFinremCaseData(nocResponse.getData(), caseType.getCcdType()))
+            .dataClassification(nocResponse.getDataClassification())
+            .securityClassification(nocResponse.getSecurityClassification())
+            .errors(nocResponse.getErrors())
+            .warnings(nocResponse.getWarnings())
+            .state(nocResponse.getState())
+            .build();
+    }
+
+
+    private void removeApplicantSolicitorDetails(FinremCaseData caseData) {
+
+        if (!caseData.isApplicantRepresentedByASolicitor()) {
+            caseData.getContactDetailsWrapper().setApplicantSolicitorName(null);
+            caseData.getContactDetailsWrapper().setApplicantSolicitorFirm(null);
+            caseData.getContactDetailsWrapper().setApplicantSolicitorAddress(null);
+            caseData.getContactDetailsWrapper().setApplicantSolicitorPhone(null);
+            caseData.getContactDetailsWrapper().setApplicantSolicitorEmail(null);
+            caseData.getContactDetailsWrapper().setApplicantSolicitorDxNumber(null);
+            caseData.getContactDetailsWrapper().setApplicantSolicitorConsentForEmails(null);
+            caseData.setApplicantOrganisationPolicy(null);
+        }
+    }
+
+    private void removeRespondentDetails(FinremCaseData caseData) {
+
+        if (caseData.isRespondentRepresentedByASolicitor()) {
```

Comment:
Perhaps this is a matter of style but I'd prefer to see a local ContactDetailsWrapper variable created rather than repeatedly using `caseData.getContactDetailsWrapper()`.

`ContactDetailsWrapper contactDetailsWrapper = caseData.getContactDetailsWrapper();`